test/cases/regression-old-worker-new-composer.sh

```diff
@@ -54,123 +59,362 @@ priority=${priority}
+else
+  echo No container runtime found, install podman or docker.
+  exit 2
+fi
+
+
+# Container image used for cloud provider CLI tools
+CONTAINER_IMAGE_CLOUD_TOOLS="quay.io/osbuild/cloud-tools:latest"
 
-sudo podman pull --creds "${V2_QUAY_USERNAME}":"${V2_QUAY_PASSWORD}" \
+greenprint "Pulling and running composer container for this commit"
+sudo ${CONTAINER_RUNTIME} pull --creds "${V2_QUAY_USERNAME}":"${V2_QUAY_PASSWORD}" \
      "quay.io/osbuild/osbuild-composer-ubi-pr:${CI_COMMIT_SHA}"
 
+cat <<EOF | sudo tee "/etc/osbuild-composer/osbuild-composer.toml"
+log_level = "debug"
+[koji]
+allowed_domains = [ "localhost", "client.osbuild.org" ]
+ca = "/etc/osbuild-composer/ca-crt.pem"
+[koji.aws_config]
+bucket = "${AWS_BUCKET}"
+[worker]
+allowed_domains = [ "localhost", "worker.osbuild.org" ]
+ca = "/etc/osbuild-composer/ca-crt.pem"
+EOF
+
 # The host entitlement doesn't get picked up by composer
 # see https://github.com/osbuild/osbuild-composer/issues/1845
-sudo podman run  \
+sudo ${CONTAINER_RUNTIME} run  \
      --name=composer \
      -d \
      -v /etc/osbuild-composer:/etc/osbuild-composer:Z \
      -v /etc/rhsm:/etc/rhsm:Z \
      -v /etc/pki/entitlement:/etc/pki/entitlement:Z \
      -v "$REPOS/repositories":/usr/share/osbuild-composer/repositories:Z \
-     -v "$WELDR_DIR:/run/weldr/":Z \
-     -v "$DNF_DIR:/run/osbuild-dnf-json/":Z \
-     -e OVERWRITE_CACHE_DIR="/var/cache/dnf-json" \
      -p 8700:8700 \
+     -p 8080:8080 \
      "quay.io/osbuild/osbuild-composer-ubi-pr:${CI_COMMIT_SHA}" \
-     --weldr-api --dnf-json --remote-worker-api \
-     --no-local-worker-api --no-composer-api
+     --remote-worker-api --no-local-worker-api
+
+greenprint "Wait for composer API"
+while ! openapi=$(curl  --silent  --show-error  --cacert /etc/osbuild-composer/ca-crt.pem  --key /etc/osbuild-composer/client-key.pem  --cert /etc/osbuild-composer/client-crt.pem  https://localhost:8080/api/image-builder-composer/v2/openapi); do
+    sleep 10
+done
+jq . <<< "${openapi}"
 
-# try starting a worker
+
+greenprint "Starting osbuild-remote-worker service and dnf-json socket"
 set +e
-sudo systemctl start osbuild-remote-worker@localhost:8700.service
+# reload in case there were changes in units
+sudo systemctl daemon-reload
+sudo systemctl enable --now osbuild-remote-worker@localhost:8700.service
 while ! sudo systemctl --quiet is-active osbuild-remote-worker@localhost:8700.service; do
     sudo systemctl status osbuild-remote-worker@localhost:8700.service
     sleep 1
-    sudo systemctl start osbuild-remote-worker@localhost:8700.service
+    sudo systemctl enable --now osbuild-remote-worker@localhost:8700.service
+done
+sudo systemctl enable --now osbuild-dnf-json.socket
+while ! sudo systemctl --quiet is-active osbuild-dnf-json.socket; do
+    sudo systemctl status osbuild-dnf-json.socket
+    sleep 1
+    sudo systemctl enable --now osbuild-dnf-json.socket
 done
 set -e
 
-function log_on_exit() {
-    sudo podman logs composer
```

Comment:
should we save the container log as an artifact?

Reply:
Indeed that would be very useful.